### ISSUE TITLE
Fix default configuration for recent Java versions

### ISSUE THREAD

**Issue by myllynen** [performancecopilot/parfait#57]
The default configuration causes exception on startup with newer Java versions:

```
# java -version
openjdk version "1.8.0_161"
OpenJDK Runtime Environment (build 1.8.0_161-b14)
OpenJDK 64-Bit Server VM (build 25.161-b14, mixed mode)
# java -javaagent:/parfait-agent-jar-with-dependencies.jar KeyboardReader
194 [main] INFO io.pcp.parfait.pcp.PcpMonitorBridge - PCP monitoring bridge started for writer [PcpMmvWriter[byteBufferFactory=FileByteBufferFactory[file=/var/lib/pcp/tmp/mmv/KeyboardReader]]]
Enter a line of input
```

```
# java -version
openjdk version "9.0.4"
OpenJDK Runtime Environment (build 9.0.4+11)
OpenJDK 64-Bit Server VM (build 9.0.4+11, mixed mode)
# java -javaagent:/parfait-agent-jar-with-dependencies.jar KeyboardReader
[main] ERROR io.pcp.parfait.ParfaitAgent  - Stopping Parfait agent [KeyboardReader]
java.lang.RuntimeException: Metric java.memorypool.eden[init] is not optional but has bad attribute [Usage]
	at io.pcp.parfait.AgentMonitoringView.createMonitorable(AgentMonitoringView.java:108)
	at io.pcp.parfait.AgentMonitoringView.register(AgentMonitoringView.java:92)
	at io.pcp.parfait.ParfaitAgent.startView(ParfaitAgent.java:119)
...
```

Tested on Fedora 27.

**Comment by keilw** [comment]
I can't say, if a different configuration helps, but if parfait-dropwizard is run with Java 9, ParfaitReporterTest fails:
```
org.mockito.exceptions.base.MockitoException: 
Cannot mock/spy class java.util.concurrent.TimeUnit
Mockito cannot mock/spy because :
 - final class
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:49)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.mockito.runners.MockitoJUnitRunner.run(MockitoJUnitRunner.java:53)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:542)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:770)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:464)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
```

**Comment by tallpsmith** [comment]
We can use the Maven Enforcer plugin to require JDK 9 or lower - I get burnt by this myself all the time!

**Comment by tallpsmith** [comment]
ok we have  2 issues here, one is runtime, one is is build time.

Runtime - @myllynen As each Java version changes, the JMX namespace for the Memory pools seems to shift; never the same JMX FQDN it seems...  Not easy to sniff these out as yet

Build time - @keilw this is trickier because Mockito can't touch Final classes...

I've set the Enforcer plugin to mandate Java8 for now - at least we get a fail fast during the build and a reason for it.  This addresses the build time, but not the runtime.

**Comment by natoscott** [comment]
This is fixed by #86
